### PR TITLE
fix: add getter methods for managed identity

### PR DESCRIPTION
**Pull request by tariq1890**
Adding helper methods to increase reuse of code.

### CODE REVIEW

**Comment by jackfrancis** [comment]
Can we unit test these?

**Comment by tariq1890** [comment]
@jackfrancis Sure thing!

File: pkg/api/types.go [hunk 1]
```diff
@@ -1290,6 +1290,32 @@ func (k *KubernetesConfig) IsRBACEnabled() bool {
 	return false
 }
 
+// UserAssignedIDEnabled checks if the user assigned ID is enabled or not.
+func (k *KubernetesConfig) UserAssignedIDEnabled() bool {
```
Comment (CecileRobertMichon) [inline pkg/api/types.go]: Can you incorporate these new functions in existing unit tests if they aren't already getting covered?

Comment (tariq1890) [reply pkg/api/types.go]: Sure @CecileRobertMichon . I will put out another PR to start doing that.

Comment (CecileRobertMichon) [reply pkg/api/types.go]: what do you mean by "rationalize our test paths"?

File: pkg/api/types_test.go [hunk 2]
```diff
@@ -2737,3 +2737,45 @@ func TestIsFeatureEnabled(t *testing.T) {
...
+		t.Errorf("expected user assigned ID to be %s, but got %s", expected, k.GetUserAssignedID())
+	}
+
+	k = KubernetesConfig{
+		UseManagedIdentity: false,
+		UserAssignedID:     "fooID",
+	}
+
+	if k.GetUserAssignedID() != "" {
+		t.Errorf("expected user assigned ID to be empty when useManagedIdentity is set to false")
```
Comment (jackfrancis) [inline pkg/api/types_test.go]: Should we file a backlog issue to add validation for these pathological scenarios?

File: pkg/api/types.go (hunk 1 above)
Comment (jackfrancis) [inline pkg/api/types.go]: let's add a unit test for `UserAssignedIDEnabled`

File: pkg/api/types.go [hunk 3]
```diff
@@ -1290,6 +1290,32 @@ func (k *KubernetesConfig) IsRBACEnabled() bool {
...
+	return k.UseManagedIdentity && k.UserAssignedID != ""
+}
+
+// UserAssignedClientIDEnabled checks if the user assigned client ID is enabled or not.
+func (k *KubernetesConfig) UserAssignedClientIDEnabled() bool {
```
Comment (jackfrancis) [inline pkg/api/types.go]: let's add a unit test for `UserAssignedClientIDEnabled`

File: pkg/api/types_test.go (hunk 2 above)
Comment (tariq1890) [reply pkg/api/types_test.go]: Validation is already present. This test case just encompasses the behaviour of the method that we had just migrated in this PR

Comment (jackfrancis) [reply pkg/api/types_test.go]: cool, thanks for confirming!

**Pull request merged**